src/modules/test/RecipeInvoker.cpp

```diff
@@ -80,13 +78,12 @@ void RecipeInvoker::TestBody()
             if (m_recipe.m_payload.size())
             {
                 JSON_Value *recipe_payload = json_parse_string(m_recipe.m_payload.c_str());
-                JSON_Value *returned_payload = json_parse_string(payload);
+                JSON_Value *returned_payload = json_parse_string(payloadString.c_str());
 
-                std::string recipe_payload_str(payload, payloadSize);
                 EXPECT_NE(nullptr, recipe_payload) << "Failed to parse recipe payload" << std::endl << "JSON: " << m_recipe.m_payload.c_str() << std::endl;
-                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << recipe_payload_str.c_str() << std::endl;
+                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << payloadString.c_str() << std::endl;
                 EXPECT_EQ(json_value_get_type(returned_payload), json_value_get_type(recipe_payload)) << "Non matching payload types. Recipe payload: " << json_value_get_type(recipe_payload) << ", returned payload: " << json_value_get_type(returned_payload) << std::endl;
-                EXPECT_EQ(json_value_equals(recipe_payload, returned_payload), 1) << "Non matching recipe payload" << std::endl << "Recipe   payload: " << m_recipe.m_payload << std::endl << "Returned payload: " << recipe_payload_str.c_str() << std::endl;
+                EXPECT_EQ(json_value_equals(recipe_payload, returned_payload), 1) << "Non matching recipe payload" << std::endl << "Recipe   payload: " << m_recipe.m_payload << std::endl << "Returned payload: " << payloadString.c_str() << std::endl;
```

Comment:
If the parsing fails (`returned_payload` and/or `recipe_payload` is null), what is the expected behavior of line 85, 86? I believe passing null to `json_value_get_type` or `json_value_equals` is undefined behavior for parson (could cause crashes).

Reply:
neither should crash but if the above fails, 85, 86 will also fail